library/Icingadb/Common/Database.php

```diff
@@ -37,6 +39,17 @@ public function getDb(): Connection
             ];
 
             $this->db = new Connection($config);
+
+            $adapter = $this->db->getAdapter();
+            if ($adapter instanceof Pgsql) {
+                // user is a reserved key word in PostgreSQL, so we need to quote it.
+                $quoted = $adapter->quoteIdentifier('user');
+                $this->db->getQueryBuilder()->on(QueryBuilder::ON_SELECT_ASSEMBLED, function (&$sql) use ($quoted) {
+                    $sql = str_replace(' user ', sprintf(' %s ', $quoted), $sql);
+                    $sql = str_replace(' user.', sprintf(' %s.', $quoted), $sql);
+                    $sql = str_replace('(user.', sprintf('(%s.', $quoted), $sql);
```

Comment:
Added a TODO as it's pretty hackish of course. However, `.user.` is not a problem since it is only an issue when `user` is used as a "single" identifier and not somewhere in between.